File: 1-two-sum/two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Hashtable<Integer, Integer> numTable= new Hashtable<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (numTable.containsKey(diff))
                return new int[] {numTable.get(diff), i};
            numTable.put(nums[i], i);
        }
        return null;
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_table = {}
        for i, num in enumerate(nums):
            diff = target - num
            if diff in num_table:
                return [num_table[diff], i]
            num_table[num] = i
        return None